scraper/src/kolibri2zim/scraper.py

```diff
@@ -196,29 +202,53 @@ def get_or_create_node_slug(self, node) -> str:
         """Compute a unique slug to be used as URL for a given node"""
         if node["id"] in self.nodes_ids_to_slugs:
             return self.nodes_ids_to_slugs[node["id"]]
-        slug = slugify(str(node.get("title", node["id"])))
+        if "title" in node:
+            slug = f"{slugify(node['title'])}-{node['id'][:4]}"
+        else:
+            slug = node["id"]
         if slug in self.nodes_ids_to_slugs.values():
-            suffix = 1
-            while True:
-                if f"{slug}_{suffix}" not in self.nodes_ids_to_slugs.values():
-                    break
-                suffix += 1
-            slug = f"{slug}_{suffix}"
+            # detect extreme case where we have a conflict
+            conflicting_node_id = {
```

Comment:
libzim already emits a warning when you try to add an entry on a path that already has an entry. Maybe that's enough? But your log message is clearer and more importantly it happens at the right moment in logs (while libzim one being async it might happen later – although it does contain the conflicting path)

Reply:
My idea is that I do not want a warning but a scraper failure, and as I you said, I want it to happen asap.
Here all slugs are computed once all topics have been explored since we grab info about children, so it is pretty soon in the process.
Probability of occurrence seems extremely low, but I do not want that it finally happens without us knowing it and developing a fix. For instance it is not clear for me if it is wise to use the first 4 chars, or if we should instead use the last 4 ones, or something else. Since UUID5 are sha1 hashes I suspect it makes no difference, but I'm not a cryptographic expert at all.

Reply:
My comment is about the log not the return (I agree about exiting). Actually scraperlib even allows you to crash on duplicates or not

Reply:
It is hard to use libzim log, since we want to exit asap and when computing slugs, we are not yet at the stage of adding items to the ZIM.
I propose to not change code.

Reply:
OK